src/test/java/teammates/test/cases/browsertests/InstructorFeedbackResultsPageUiTest.java

```diff
@@ -634,102 +628,137 @@ public void testIndicateMissingResponses() {
+        resultsPage.loadResultQuestionPanel(2);
+
+        editCommentGivenByDifferentInstructors("-1-1-0");
+
+        ______TS("Question view: Typical case: delete existing feedback response comments using comment modal");
 
         resultsPage = loginToInstructorFeedbackResultsPage("CFResultsUiT.instr", "Open Session");
         resultsPage.displayByQuestion();
         resultsPage.loadResultQuestionPanel(1);
 
-        resultsPage.clickCommentModalButton("-2-1-0");
-        resultsPage.addFeedbackResponseCommentInCommentModal("showResponseCommentAddForm-2-1-0",
-                "successive action comment");
-        resultsPage.verifyCommentRowContent("-2-1-0-1", "successive action comment", "Teammates Test");
+        deleteComment("-2-1-0");
+        deleteComment("-3-1-0");
+    }
 
-        resultsPage.editFeedbackResponseComment("-2-1-0-1", "edited successive action comment");
-        resultsPage.verifyCommentRowContent("-2-1-0-1", "edited successive action comment",
-                "Teammates Test");
-        resultsPage.clickVisibilityOptionForResponseCommentAndSave("responseCommentRow-2-1-0-1", 1);
+    private void addCommentToEmptyResponse(String commentId) {
+        resultsPage.clickCommentModalButton(commentId);
+        resultsPage.addFeedbackResponseCommentInCommentModal("showResponseCommentAddForm" + commentId, "");
+        resultsPage.verifyCommentFormErrorMessage(commentId, Const.StatusMessages.FEEDBACK_RESPONSE_COMMENT_EMPTY);
+        resultsPage.closeCommentModal(commentId);
+    }
+
+    private void addCommentValidToResponse(String commentId) {
+        String editFormCommentId = commentId + "-1";
+        resultsPage.clickCommentModalButton(commentId);
+        resultsPage.addFeedbackResponseCommentInCommentModal("showResponseCommentAddForm" + commentId, "test comment 1");
+        resultsPage.verifyCommentRowContent(editFormCommentId, "test comment 1", "Teammates Test");
+        resultsPage.isElementPresent(By.id("showResponseCommentAddForm" + commentId));
+        resultsPage.verifyContainsElement(By.id("visibility-options" + editFormCommentId));
+        resultsPage.closeCommentModal(commentId);
+    }
+
+    private void editCommentOnResponse(String commentId) {
+        String editFormCommentId = commentId + "-1";
+        resultsPage.clickCommentModalButton(commentId);
+        resultsPage.editFeedbackResponseComment(editFormCommentId, "edited test comment");
+        resultsPage.verifyCommentRowContent(editFormCommentId, "edited test comment", "Teammates Test");
+        resultsPage.isElementPresent(By.id("showResponseCommentAddForm" + commentId));
+        resultsPage.closeCommentModal(commentId);
+    }
 
-        resultsPage.deleteFeedbackResponseCommentInModal("-2-1-0-1");
-        resultsPage.isElementPresent(By.id("showResponseCommentAddForm-2-1-0"));
-        resultsPage.closeCommentModal("-2-1-0");
+    private void editCommentGivenByDifferentInstructors(String commentId) {
```

Comment:
Duplication with editCommentOnResponse?

Pass comment as a param might be a good choice.